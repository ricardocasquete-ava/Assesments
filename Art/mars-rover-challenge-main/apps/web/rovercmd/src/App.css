.App {
  text-align: center;
}

.App-header, .App-sub-header, .App-section, .App-error {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  padding: 20px 0px;
}

.App-header {
  font-size: calc(10px + 2vmin);
  padding: 20px 0px;
}

.App-sub-header {
  font-size: calc(10px + 0.5vmin);
  padding: 10px 0px;
}

.App-sub-header p {
  margin: 0px 0px 10px 0px;
}

.App-error {
  font-size: calc(10px + 1vmin);
  padding: 5px 5px;
  color: #ed143d;
}

.App-section {
  font-size: calc(10px + 1vmin);
  padding: 10px 0px;
}

.App-section p {
  margin: 10px;
}

.App a {
  color: #1e90ff;
  text-decoration: none;
}

.App a:hover {
  text-decoration: underline;
}