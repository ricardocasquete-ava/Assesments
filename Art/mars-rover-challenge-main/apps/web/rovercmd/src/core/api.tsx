import { baseUrl } from './apiConfig';

async function post<T> (url: string, data: Object): Promise<T> {
    const rawResponse = await fetch(`${baseUrl}${url}`, {
        method: 'POST',
        mode: 'cors',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(data)
    });

    return rawResponse.json() as Promise<T>;
}

export { post };