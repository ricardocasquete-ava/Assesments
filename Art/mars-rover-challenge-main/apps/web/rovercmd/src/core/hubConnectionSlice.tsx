import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import type { RootState } from './store';

interface HubConnectionState {
    connectionId: string | null
};

const initialState: HubConnectionState = {
    connectionId: null
};

export const hubConnectionSlice = createSlice({
  name: 'hubConnection',
  initialState,
  reducers: {
    setConnectionId: (state, action: PayloadAction<string | null>) => {
      state.connectionId = action.payload
    }
  },
});

export const { setConnectionId } = hubConnectionSlice.actions;

export const getConnectionId = (state: RootState) => state.hubConnection.connectionId;

export default hubConnectionSlice.reducer;