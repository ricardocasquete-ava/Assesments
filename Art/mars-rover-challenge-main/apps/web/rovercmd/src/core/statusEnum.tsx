export enum Status {
    Ready = 0,
    Busy = 1,
    Error = -1
};