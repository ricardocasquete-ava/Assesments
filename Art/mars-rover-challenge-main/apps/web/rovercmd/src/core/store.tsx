import { configureStore } from '@reduxjs/toolkit';
import roverReducer from './roverSlice';
import hubConnectionReducer from './hubConnectionSlice';

const store = configureStore({
  reducer: {
    rover: roverReducer,
    hubConnection: hubConnectionReducer
  }
});

export default store;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;