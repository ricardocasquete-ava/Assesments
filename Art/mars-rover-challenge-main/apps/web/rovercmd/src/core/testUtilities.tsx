import { render as reactRender, RenderResult } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import roverReducer from './roverSlice';
import hubConnectionReducer from './hubConnectionSlice';

function render(
  ui: React.ReactElement<any>,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        rover: roverReducer,
        hubConnection: hubConnectionReducer
      },
      preloadedState
    }),
    ...renderOptions
  }: any = {}
): RenderResult {
  function Wrapper({ children }: any) {
    return <Provider store={store}>{children}</Provider>
  }
  return reactRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }