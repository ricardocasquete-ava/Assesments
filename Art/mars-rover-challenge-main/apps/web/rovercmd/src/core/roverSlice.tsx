import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import type { AppDispatch, RootState } from './store';
import { postInput } from 'services/sendService';
import { Status } from './statusEnum';

interface RoverState {
    status: Status,
    output: string
};

const initialState: RoverState = {
    status: Status.Ready,
    output: ''
};

export const roverSlice = createSlice({
  name: 'rover',
  initialState,
  reducers: {
    setOutput: (state, action: PayloadAction<string>) => {
      state.output = action.payload
    },
    setStatus: (state, action: PayloadAction<Status>) => {
      state.status = action.payload
    }
  },
});

export const { setOutput , setStatus } = roverSlice.actions;

export const sendInput = (command: string) => (dispatch: AppDispatch) => {
  dispatch(setStatus(Status.Busy));
  dispatch(setOutput(''));

  postInput(command)
    .then(response => {
      dispatch(setOutput(response));
      dispatch(setStatus(Status.Ready));
    })
    .catch(error => {
      dispatch(setOutput(''));
      dispatch(setStatus(Status.Error));
    });
};

export const getOutput = (state: RootState) => state.rover.output;
export const getStatus = (state: RootState) => state.rover.status;

export default roverSlice.reducer;