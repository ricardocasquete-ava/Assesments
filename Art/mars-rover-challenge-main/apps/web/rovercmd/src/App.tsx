import './App.css';
import CommandInput from 'features/CommandInput/';
import CommandOutput from 'features/CommandOutput/';
import HubConnector from 'features/HubConnector';
import { useAppSelector } from 'core/hooks';
import { getStatus } from 'core/roverSlice';
import { Status } from 'core/statusEnum';

const hasError = (status: Status) => status === Status.Error;

function App() {
  const status = useAppSelector(getStatus);

  return (
    <div className="App">

      {hasError(status)
        ? <section className='App-error'>Error!</section>
        : null}

      <header className="App-header">
        Mars Rover Challenge Web App
      </header>
      <section className="App-sub-header">
        <p>A web-based wrapper for the <code>MarsRoverChallenge.Send</code> library.</p>
        <p>
          This library is an implementation of the <a href='https://code.google.com/archive/p/marsrovertechchallenge/'
            target='_blank'
            rel="noreferrer">
            Mars Rover
          </a> challenge.
        </p>
      </section>
      <section className="App-section">
        <CommandInput />
      </section>
      <section className="App-section">
        <CommandOutput />
      </section>
      <section className="App-section">
        <HubConnector />
      </section>
    </div>
  );
}

export default App;
