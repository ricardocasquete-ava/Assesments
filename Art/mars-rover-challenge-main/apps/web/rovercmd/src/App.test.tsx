import { render } from 'core/testUtilities';

jest.mock('features/HubConnector');

import App from './App';

test('renders App', () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
});
