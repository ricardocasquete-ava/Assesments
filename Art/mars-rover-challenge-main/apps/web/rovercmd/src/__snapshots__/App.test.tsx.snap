// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    Mars Rover Challenge Web App
  </header>
  <section
    class="App-sub-header"
  >
    <p>
      A web-based wrapper for the 
      <code>
        MarsRoverChallenge.Send
      </code>
       library.
    </p>
    <p>
      This library is an implementation of the 
      <a
        href="https://code.google.com/archive/p/marsrovertechchallenge/"
        rel="noreferrer"
        target="_blank"
      >
        Mars Rover
      </a>
       challenge.
    </p>
  </section>
  <section
    class="App-section"
  >
    <div
      class="CommandInput"
    >
      <fieldset>
        <legend>
          Enter command(s) to send to the rover(s):
        </legend>
        <textarea
          rows="10"
        />
      </fieldset>
      <button>
        Send!
      </button>
    </div>
  </section>
  <section
    class="App-section"
  >
    <div
      class="CommandOutput"
    >
      <fieldset>
        <legend>
          Output:
        </legend>
        <textarea
          readonly=""
          rows="5"
        />
      </fieldset>
    </div>
  </section>
  <section
    class="App-section"
  >
    <div
      class="HubConnector"
    >
      Mock Hub Connector
    </div>
  </section>
</div>
`;
